File: TugasBesar/Mobil.py
import requests
import json
class Mobil:
    def __init__(self):
        self.__id=None
        self.__plat = None
        self.__merk = None
        self.__jenis = None
        self.__warna = None
        self.__hargasewa = None
        self.__url = "http://f0835423.xsph.ru/apprental/mobil_api.php"
                    
    @property
    def id(self):
        return self.__id
    @property
    def plat(self):
        return self.__plat
        
    @plat.setter
    def plat(self, value):
        self.__plat = value
    @property
    def merk(self):
        return self.__merk
        
    @merk.setter
    def merk(self, value):
        self.__merk = value
    @property
    def jenis(self):
        return self.__jenis
        
    @jenis.setter
    def jenis(self, value):
        self.__jenis = value
    @property
    def warna(self):
        return self.__warna
        
    @warna.setter
    def warna(self, value):
        self.__warna = value
    @property
    def hargasewa(self):
        return self.__hargasewa
        
    @hargasewa.setter
    def hargasewa(self, value):
        self.__hargasewa = value
    def get_all(self):
        payload ={}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(self.__url, json=payload, headers=headers)
        return response.text
    def get_by_plat(self, plat):
        url = self.__url+"?plat="+plat
        payload = {}
        headers = {'Content-Type': 'application/json'}
        response = requests.get(url, json=payload, headers=headers)
        data = json.loads(response.text)
        for item in data:
            self.__id = item['id']
            self.__plat = item['plat']
            self.__merk = item['merk']
            self.__jenis = item['jenis']
            self.__warna = item['warna']
            self.__hargasewa = item['hargasewa']
        return data
    def simpan(self):
        payload = {
            "plat":self.__plat,
            "merk":self.__merk,
            "jenis":self.__jenis,
            "warna":self.__warna,
            "hargasewa":self.__hargasewa
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.post(self.__url, data=payload, headers=headers)
        return response.text
    def update_by_plat(self, plat):
        url = self.__url+"?plat="+plat
        payload = {
            "plat":self.__plat,
            "merk":self.__merk,
            "jenis":self.__jenis,
            "warna":self.__warna,
            "hargasewa":self.__hargasewa
            }
        headers = {'Content-Type': 'application/x-www-form-urlencoded'}
        response = requests.put(url, data=payload, headers=headers)
        return response.text
    def delete_by_plat(self,plat):
        url = self.__url+"?plat="+plat
        headers = {'Content-Type': 'application/json'}
        payload={}
        response = requests.delete(url, json=payload, headers=headers)
        return response.text

File: TugasBesar/Mobil.php
<?php
//Simpanlah dengan nama file : Mobil.php
require_once 'database.php';
class Mobil 
{
    private $db;
    private $table = 'mobil';
    public $plat = "";
    public $merk = "";
    public $jenis = "";
    public $warna = "";
    public $hargasewa = "";
    public function __construct(MySQLDatabase $db)
    {
        $this->db = $db;
    }
    public function get_all() 
    {
        $query = "SELECT * FROM $this->table";
        $result_set = $this->db->query($query);
        return $result_set;
    }
    public function get_by_id(int $id)
    {
        $query = "SELECT * FROM $this->table WHERE id = $id";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function get_by_plat(int $plat)
    {
        $query = "SELECT * FROM $this->table WHERE plat = $plat";
        $result_set = $this->db->query($query);   
        return $result_set;
    }
    public function insert(): int
    {
        $query = "INSERT INTO $this->table (`plat`,`merk`,`jenis`,`warna`,`hargasewa`) VALUES ('$this->plat','$this->merk','$this->jenis','$this->warna','$this->hargasewa')";
        $this->db->query($query);
        return $this->db->insert_id();
    }
    public function update(int $id): int
    {
        $query = "UPDATE $this->table SET plat = '$this->plat', merk = '$this->merk', jenis = '$this->jenis', warna = '$this->warna', hargasewa = '$this->hargasewa' 
        WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function update_by_plat($plat): int
    {
        $query = "UPDATE $this->table SET plat = '$this->plat', merk = '$this->merk', jenis = '$this->jenis', warna = '$this->warna', hargasewa = '$this->hargasewa' 
        WHERE plat = $plat";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete(int $id): int
    {
        $query = "DELETE FROM $this->table WHERE id = $id";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
    public function delete_by_plat($plat): int
    {
        $query = "DELETE FROM $this->table WHERE plat = $plat";
        $this->db->query($query);
        return $this->db->affected_rows();
    }
}
?>